Importing k8snodepool is not working

## Description
When trying to import an existing k8snodepool in terraform, the plugin crashes

## Expected behavior
k8snodepool is successfully imported

## Environment

Terraform version: 
```
Terraform v1.0.4
on linux_amd64
+ provider registry.terraform.io/ionos-cloud/ionoscloud v6.0.0-beta.4
```


## Configuration Files
```

resource "ionoscloud_k8s_cluster" "example" {
  k8s_version = "1.20.8"
  name        = "my-test-k8s"
  public      = true
  maintenance_window {
    day_of_the_week = "Sunday"
    time            = "01:02:39Z"
  }
}

resource "ionoscloud_k8s_node_pool" "example" {
}

```

## How to Reproduce

`terraform import ionoscloud_k8s_node_pool.example fda7e7d8-5075-48c5-b760-28087e09bf3c/3da10089-5e83-4e91-b1b4-897e6f7a1d25
`
## Error and Debug Output
```
ionoscloud_k8s_node_pool.example: Importing from ID "fda7e7d8-5075-48c5-b760-28087e09bf3c/3da10089-5e83-4e91-b1b4-897e6f7a1d25"...
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ImportResourceState call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-ionoscloud_v6.0.0-beta.4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10dac48]

goroutine 35 [running]:
github.com/ionos-cloud/terraform-provider-ionoscloud/ionoscloud.resourceK8sNodepoolImport(0x1572a28, 0xc0002a80c0, 0xc000198680, 0x129ad80, 0xc0004105b0, 0x1, 0x30, 0x7f2080755a68, 0x30, 0xc000558390)
	github.com/ionos-cloud/terraform-provider-ionoscloud/ionoscloud/utils.go:161 +0xc08
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000094910, 0x1572a28, 0xc0002a80c0, 0xc000627aa8, 0xc000192050, 0x49, 0x203000, 0x203000, 0x203000, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/provider.go:367 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc00000c5e8, 0x1572a28, 0xc0002a80c0, 0xc0004b40a0, 0xc0002a80c0, 0x12fd160, 0xc000558100)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:1021 +0xe5
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ImportResourceState(0xc0006c8600, 0x1572ad0, 0xc0002a80c0, 0xc00055a0a0, 0xc0006c8600, 0xc000558180, 0xc00034aba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:349 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler(0x12fd160, 0xc0006c8600, 0x1572ad0, 0xc000558180, 0xc0004fe180, 0x0, 0x1572ad0, 0xc000558180, 0xc000088150, 0x65)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:398 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000236a80, 0x157c218, 0xc000103c80, 0xc000126100, 0xc000691530, 0x1c2c4d8, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000236a80, 0x157c218, 0xc000103c80, 0xc000126100, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00028c1a0, 0xc000236a80, 0x157c218, 0xc000103c80, 0xc000126100)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ionoscloud_v6.0.0-beta.4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

## Additional Notes
Tried also with terraform v0.15.0 same error
Tried with older version of the provider v5.2.4, v5.2.0
Maybe smth change in the api?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 6.0.0-beta.5
+
+- issue #77 - fix import for k8s nodepool
+
 ## 6.0.0-beta.4
 
 - fix: issue #71 - recreate nodepool on change of specifications

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 ## 6.0.0-beta.5
 
+- rollback to the node pool behaviour before the fix of issue #71
 - issue #77 - fix import for k8s nodepool
 
 ## 6.0.0-beta.4

--- a/docs/resources/k8s_node_pool.md
+++ b/docs/resources/k8s_node_pool.md
@@ -80,7 +80,3 @@ terraform import ionoscloud_k8s_node_pool.demo {k8s_cluster_uuid}/{k8s_nodepool_
 ```
 
 This can be helpful when you want to import kubernetes node pools which you have already created manually or using other means, outside of terraform, towards the goal of managing them via Terraform
-
-## Notes
-
-Please note that every time the arguments of the nodepool are modified, the resource is destroyed and recreated.

--- a/ionoscloud/resource_k8s_node_pool.go
+++ b/ionoscloud/resource_k8s_node_pool.go
@@ -18,7 +18,7 @@ func resourcek8sNodePool() *schema.Resource {
 	return &schema.Resource{
 		CreateContext: resourcek8sNodePoolCreate,
 		ReadContext:   resourcek8sNodePoolRead,
-		UpdateContext: nil,
+		UpdateContext: resourcek8sNodePoolUpdate,
 		DeleteContext: resourcek8sNodePoolDelete,
 		Importer: &schema.ResourceImporter{
 			StateContext: resourceK8sNodepoolImport,
@@ -29,7 +29,6 @@ func resourcek8sNodePool() *schema.Resource {
 				Description:  "The desired name for the node pool",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"k8s_version": {
 				Type:        schema.TypeString,
@@ -52,7 +51,6 @@ func resourcek8sNodePool() *schema.Resource {
 					}
 					return false
 				},
-				ForceNew: true,
 			},
 			"auto_scaling": {
 				Type:        schema.TypeList,
@@ -73,7 +71,6 @@ func resourcek8sNodePool() *schema.Resource {
 						},
 					},
 				},
-				ForceNew: true,
 			},
 			"lans": {
 				Type:        schema.TypeList,
@@ -113,7 +110,6 @@ func resourcek8sNodePool() *schema.Resource {
 						},
 					},
 				},
-				ForceNew: true,
 			},
 			"maintenance_window": {
 				Type:        schema.TypeList,
@@ -134,66 +130,56 @@ func resourcek8sNodePool() *schema.Resource {
 						},
 					},
 				},
-				ForceNew: true,
 			},
 			"datacenter_id": {
 				Type:         schema.TypeString,
 				Description:  "The UUID of the VDC",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"k8s_cluster_id": {
 				Type:         schema.TypeString,
 				Description:  "The UUID of an existing kubernetes cluster",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"cpu_family": {
 				Type:         schema.TypeString,
 				Description:  "CPU Family",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"availability_zone": {
 				Type:         schema.TypeString,
 				Description:  "The compute availability zone in which the nodes should exist",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"storage_type": {
 				Type:         schema.TypeString,
 				Description:  "Storage type to use",
 				Required:     true,
 				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
-				ForceNew:     true,
 			},
 			"node_count": {
 				Type:        schema.TypeInt,
 				Description: "The number of nodes in this node pool",
 				Required:    true,
-				ForceNew:    true,
 			},
 			"cores_count": {
 				Type:        schema.TypeInt,
 				Description: "CPU cores count",
 				Required:    true,
-				ForceNew:    true,
 			},
 			"ram_size": {
 				Type:        schema.TypeInt,
 				Description: "The amount of RAM in MB",
 				Required:    true,
-				ForceNew:    true,
 			},
 			"storage_size": {
 				Type:        schema.TypeInt,
 				Description: "The total allocated storage capacity of a node in GB",
 				Required:    true,
-				ForceNew:    true,
 			},
 			"public_ips": {
 				Type:        schema.TypeList,
@@ -202,7 +188,6 @@ func resourcek8sNodePool() *schema.Resource {
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
-				ForceNew: true,
 			},
 		},
 		Timeouts: &resourceDefaultTimeouts,

--- a/ionoscloud/resource_k8s_node_pool_test.go
+++ b/ionoscloud/resource_k8s_node_pool_test.go
@@ -51,10 +51,10 @@ func TestAcck8sNodepool_Basic(t *testing.T) {
 				),
 			},
 			{
-				Config: fmt.Sprintf(testAccCheckk8sNodepoolConfigUpdate, publicIp1, publicIp2, publicIp3),
+				Config: fmt.Sprintf(testAccCheckk8sNodepoolConfigUpdate, k8sNodepoolName, publicIp1, publicIp2, publicIp3),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckk8sNodepoolExists("ionoscloud_k8s_node_pool.terraform_acctest", &k8sNodepool),
-					resource.TestCheckResourceAttr("ionoscloud_k8s_node_pool.terraform_acctest", "name", "updated"),
+					resource.TestCheckResourceAttr("ionoscloud_k8s_node_pool.terraform_acctest", "name", k8sNodepoolName),
 					resource.TestCheckResourceAttr("ionoscloud_k8s_node_pool.terraform_acctest", "public_ips.0", publicIp1),
 					resource.TestCheckResourceAttr("ionoscloud_k8s_node_pool.terraform_acctest", "public_ips.1", publicIp2),
 					resource.TestCheckResourceAttr("ionoscloud_k8s_node_pool.terraform_acctest", "public_ips.2", publicIp3),
@@ -247,7 +247,7 @@ resource "ionoscloud_k8s_cluster" "terraform_acctest" {
 }
 
 resource "ionoscloud_k8s_node_pool" "terraform_acctest" {
-  name        = "updated"
+  name        = "%s"
   k8s_version = "${ionoscloud_k8s_cluster.terraform_acctest.k8s_version}"
   auto_scaling {
   	min_node_count = 1

--- a/ionoscloud/utils.go
+++ b/ionoscloud/utils.go
@@ -158,7 +158,7 @@ func resourceK8sNodepoolImport(ctx context.Context, d *schema.ResourceData, meta
 		log.Printf("[INFO] Setting Public IPs for k8s node pool %s to %+v...", d.Id(), d.Get("public_ips"))
 	}
 
-	if k8sNodepool.Properties.AutoScaling != nil && (*k8sNodepool.Properties.AutoScaling.MinNodeCount != 0 && *k8sNodepool.Properties.AutoScaling.MaxNodeCount != 0) {
+	if k8sNodepool.Properties.AutoScaling != nil && k8sNodepool.Properties.AutoScaling.MinNodeCount != nil && k8sNodepool.Properties.AutoScaling.MaxNodeCount != nil && (*k8sNodepool.Properties.AutoScaling.MinNodeCount != 0 && *k8sNodepool.Properties.AutoScaling.MaxNodeCount != 0) {
 		if err := d.Set("auto_scaling", []map[string]int32{
 			{
 				"min_node_count": *k8sNodepool.Properties.AutoScaling.MinNodeCount,

